api/v1alpha1/timeout_types.go

```diff
@@ -0,0 +1,26 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+
+// ConnectionTimeouts defines configuration for timeouts related to connections.
+type ConnectionTimeouts struct {
+	// The timeout for new network connection establishment, including TCP and TLS handshakes.
+	//
```

Comment:
should we create top level `tcp` and `http` fields here ?

Reply:
Seeing as there are several other HTTP (stream, ...) and TCP (proxy, protocol, ...) timeouts that could be relevant in the future, it makes sense to me.